x-pack/plugins/security_solution/server/endpoint/routes/actions/file_download_handler.ts

```diff
@@ -61,22 +65,23 @@ export const getActionFileDownloadRouteHandler = (
   const logger = endpointContext.logFactory.get('actionFileDownload');
 
   return async (context, req, res) => {
-    const fleetFiles = await endpointContext.service.getFleetFromHostFilesClient();
-    const esClient = (await context.core).elasticsearch.client.asInternalUser;
     const { action_id: actionId, file_id: fileId } = req.params;
 
     try {
-      await validateActionId(esClient, actionId);
-      const file = await fleetFiles.get(fileId);
-
-      if (file.id !== fileId) {
-        throw new CustomHttpRequestError(
-          `Invalid file id [${fileId}] for action [${actionId}]`,
-          400
-        );
-      }
+      const esClient = (await context.core).elasticsearch.client.asInternalUser;
+      const { agentType } = await getActionAgentType(esClient, actionId);
+      const user = endpointContext.service.security?.authc.getCurrentUser(req);
+      const casesClient = await endpointContext.service.getCasesClient(req);
+      const connectorActions = (await context.actions).getActionsClient();
+      const responseActionsClient: ResponseActionsClient = getResponseActionsClient(agentType, {
+        esClient,
+        casesClient,
+        endpointService: endpointContext.service,
+        username: user?.username || 'unknown',
```

Comment:
This condition should never happen since we run Kibana with security enabled, so this is just a safety net. That being said, I think that having a value allows us to identify who - or more accurately in this case, "where" the update came from. its more for debugging and tracking. It also ensures that we have an actual value to be returned in API calls